App/xcbuddykit/Sources/Generator/ConfigGenerator.swift

```diff
@@ -118,36 +125,34 @@ final class ConfigGenerator: ConfigGenerating {
     private func generateProjectSettingsFor(buildConfiguration: BuildConfiguration,
                                             configuration: Configuration?,
                                             project: Project,
-                                            groups: ProjectGroups,
-                                            sourceRootPath: AbsolutePath,
-                                            context: GeneratorContexting,
+                                            groups _: ProjectGroups,
+                                            fileElements: ProjectFileElements,
+                                            sourceRootPath _: AbsolutePath,
+                                            context _: GeneratorContexting,
                                             pbxproj: PBXProj,
                                             configurationList: XCConfigurationList) throws {
         let variant: BuildSettingsProvider.Variant = (buildConfiguration == .debug) ? .debug : .release
         let defaultConfigSettings = BuildSettingsProvider.projectDefault(variant: variant)
         let defaultSettingsAll = BuildSettingsProvider.projectDefault(variant: .all)
-        
+
         var settings: [String: Any] = [:]
         extend(buildSettings: &settings, with: defaultSettingsAll)
         extend(buildSettings: &settings, with: project.settings?.base ?? [:])
         extend(buildSettings: &settings, with: defaultConfigSettings)
-        
+
         let variantBuildConfiguration = XCBuildConfiguration(name: buildConfiguration.rawValue.capitalized)
         if let variantConfig = configuration {
             extend(buildSettings: &settings, with: variantConfig.settings)
             if let xcconfig = variantConfig.xcconfig {
-                let fileReference = try fileGenerator.generateFile(path: xcconfig,
-                                                                   in: groups.projectConfigurations(),
-                                                                   sourceRootPath: sourceRootPath,
-                                                                   context: context)
-                variantBuildConfiguration.baseConfigurationReference = fileReference.reference
+                let fileReference = fileElements.file(path: xcconfig)
```

Comment:
The good thing is that if any other generator needs the file, we can just get it by passing the absolute path 😛